src/Streams.jl

```diff
@@ -202,13 +202,20 @@ function Base.eof(http::Stream)
     if !headerscomplete(http.message)
         startread(http)
     end
+    if http.ntoread == unknown_length
+        return eof(http.stream)
+    end
     if http.ntoread == 0
         return true
     end
     if eof(http.stream)
-        return true
+        # ntoread is neither 0 nor unknown_length here
+        # so eof was reached before fixed content-length delivered
+        @debug 1 "❗️  EOFError: Transfer closed with $(http.ntoread) bytes remaining to read   $(http.stream)"
```

Comment:
Does `http.stream` contain anything useful?
```suggestion
        @debug 1 "❗️  EOFError: Transfer closed with $(http.ntoread) bytes remaining to read"
```